### PR TITLE
[Develop] Feature/feature edit personal information

### PR DESCRIPTION
**Pull request by samuelloza**

### CODE REVIEW

File: salesforce-gui/src/main/java/org/fundacionjala/salesforce/ui/commonPages/accountPage/AccountEditPage.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class AccountEditPage extends BasePage {
```
Comment (MirkoFer-622) [inline salesforce-gui/src/main/java/org/fundacionjala/salesforce/ui/commonPages/accountPage/AccountEditPage.java:16]: The name could be EditAccountPage, plaese change it

Comment (MirkoFer-622) [reply salesforce-gui/src/main/java/org/fundacionjala/salesforce/ui/commonPages/accountPage/AccountEditPage.java:16]: And add [SL] description...

File: salesforce-gui/src/main/java/org/fundacionjala/salesforce/ui/context/Context.java [hunk 2]
```diff
@@ -11,21 +13,43 @@
...
         this.usersList = new ArrayList<>();
     }
 
+    /**
+     * Sets request specification.
+     *
+     * @param requestSpe
+     */
+    public void setRequestSpec(final RequestSpecification requestSpe) {
+        requestManager.setRequestSpec(requestSpe);
```
Comment (MirkoFer-622) [inline salesforce-gui/src/main/java/org/fundacionjala/salesforce/ui/context/Context.java]: This line is already present in the Login Step, Is it necesary into this class again??

File: salesforce-gui/src/main/java/org/fundacionjala/salesforce/ui/entities/PersonalInformation.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
+package org.fundacionjala.salesforce.ui.entities;
+
+/**
+ * [SL] Entity that represents a Personal Information in Salesforce.
+ */
+public class PersonalInformation {
```
Comment (MirkoFer-622) [inline salesforce-gui/src/main/java/org/fundacionjala/salesforce/ui/entities/PersonalInformation.java:6]: This entity is already represented by User entity, because the personal information edited is about the loged id User.

File: salesforce-gui/src/test/java/org/fundacionjala/salesforce/cucumber/hooks/PersonalInformation.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+     */
+    public PersonalInformation(final Context dataContext) {
+        this.context = dataContext;
+    }
+
+    /**
+     * Creates thee board before execute the step with the createBoard ta'g.
+     */
+    @After(value = "@restorePersonalInformation", order = 1)
+    public void createOrganization() {
```
Comment (MirkoFer-622) [inline salesforce-gui/src/test/java/org/fundacionjala/salesforce/cucumber/hooks/PersonalInformation.java]: Why the name of this method is createOrganization? And review the javadoc please

File: salesforce-gui/src/test/java/org/fundacionjala/salesforce/cucumber/stepdefs/LoginSteps.java [hunk 5]
```diff
@@ -41,15 +44,15 @@ public LoginSteps(final Context sharedContext) {
...
+        TestExecutionProperties.setRootPath("../salesforce-core/");
+        TestPropertiesSetter.setDataProviderThreadCountProp();
+        TestPropertiesSetter.setTestBrowser();
 
         //Updating User Entity to get credentials
         user = context.getUserByAlias(userAlias);
 
         //Set User Authentication to use Salesforce API in next steps
-        RequestManager.setRequestSpec(ApiAuthenticator.getLoggedReqSpec(user));
+        context.setRequestSpec(ApiAuthenticator.getLoggedReqSpec(user));
```
Comment (MirkoFer-622) [inline salesforce-gui/src/test/java/org/fundacionjala/salesforce/cucumber/stepdefs/LoginSteps.java]: Why did you carry this method to the context?? Is that its responsability?? (Remember that this set "LoggedReqSpec")

File: salesforce-gui/src/test/java/org/fundacionjala/salesforce/cucumber/stepdefs/personalInformation/EditStep.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Then;
+import org.fundacionjala.salesforce.ui.skins.SkinManager;
+import org.fundacionjala.salesforce.ui.skins.abstractPage.personalInformation.AbstractEditPersonalInformation;
+
+import java.util.Map;
+
+import org.testng.Assert;
+
+public class EditStep {
```
Comment (MirkoFer-622) [inline salesforce-gui/src/test/java/org/fundacionjala/salesforce/cucumber/stepdefs/personalInformation/EditStep.java]: Plaese, could you change this method for PersonalInformationSteps? because only edit is too generic

File: salesforce-gui/src/main/java/org/fundacionjala/salesforce/ui/skins/abstractPage/personalInformation/AbstractEditPersonalInformation.java [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public abstract class AbstractEditPersonalInformation extends BasePage {
```
Comment (MirkoFer-622) [inline salesforce-gui/src/main/java/org/fundacionjala/salesforce/ui/skins/abstractPage/personalInformation/AbstractEditPersonalInformation.java]: Please, when you have Pages, the class should end in "Page" to avoid confusions with "Form, Popup, or another component"

**Review by MirkoFer-622 — CHANGES_REQUESTED** [review]
Thanks take account the comments that I left, and not forget to add comments to all classes you created starting with "[SL]...". Thank you! 👍🏼

File: salesforce-gui/src/main/java/org/fundacionjala/salesforce/ui/skins/abstractPage/personalInformation/AbstractEditPersonalInformation.java (hunk 7 above)
Comment (samuelloza) [reply salesforce-gui/src/main/java/org/fundacionjala/salesforce/ui/skins/abstractPage/personalInformation/AbstractEditPersonalInformation.java]: Ok, thks for you comment.

File: salesforce-gui/src/test/java/org/fundacionjala/salesforce/cucumber/hooks/Hooks.java [hunk 8]
```diff
@@ -15,6 +16,18 @@
  * [SL] This class contains hooks for the project.
  */
 public class Hooks {
+
+    private Context context;
```
Comment (MirkoFer-622) [inline salesforce-gui/src/test/java/org/fundacionjala/salesforce/cucumber/hooks/Hooks.java]: Why did you add the Context?? Is it necesary here?

File: salesforce-gui/src/test/resources/features/personalInformation/edit_personalInformation.feature [hunk 9]
```diff
@@ -2,15 +2,13 @@
 Feature: Edit Personal Information
 
   Background:
-    Given I log in to Salesforce with "Editable User" credentials
+    Given I log in to Salesforce with "Account Owner User" credentials
```
Comment (MirkoFer-622) [inline salesforce-gui/src/test/resources/features/personalInformation/edit_personalInformation.feature:5]: I think that "Account Owner User" could be a different kind of user related to scenarios where we work with Accounts, please change to "Editable User" because should be different Users and add him to InitialsUsers.json file.

File: salesforce-gui/src/main/java/org/fundacionjala/salesforce/ui/entities/PersonalInformation.java (hunk 3 above)
Comment (MirkoFer-622) [reply salesforce-gui/src/main/java/org/fundacionjala/salesforce/ui/entities/PersonalInformation.java:6]: I think you should work with "User" entity, please remove this entity.

File: salesforce-gui/src/main/java/org/fundacionjala/salesforce/ui/context/Context.java (hunk 2 above)
Comment (MirkoFer-622) [reply salesforce-gui/src/main/java/org/fundacionjala/salesforce/ui/context/Context.java]: Please, this line is repeated in LoginSteps class, please remove this method from Context

**Review by MirkoFer-622 — CHANGES_REQUESTED** [review]
Please delete PersonalInformation entity, thi scenario is related to the USer entity, we are editing the User entity.

File: salesforce-gui/src/test/java/org/fundacionjala/salesforce/cucumber/stepdefs/personalInformation/PersonalInformationSteps.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+public class PersonalInformationSteps {
+
+    private AbstractEditPersonalInformationPage personalInformation;
+
+    /**
+     * [SL] Edits the personal information.
+     *
+     * @param formData
+     */
+    @And("I edit my personal information with the following data")
```
Comment (MirkoFer-622) [inline salesforce-gui/src/test/java/org/fundacionjala/salesforce/cucumber/stepdefs/personalInformation/PersonalInformationSteps.java]: Avoid use @And please, replace it with @When

File: salesforce-gui/src/test/java/org/fundacionjala/salesforce/cucumber/hooks/Hooks.java (hunk 8 above)
Comment (samuelloza) [reply salesforce-gui/src/test/java/org/fundacionjala/salesforce/cucumber/hooks/Hooks.java]: Ok, I delete these line unnecessaries

**Comment by MirkoFer-622** [comment]
LGTM!
